doseCalc/+DoseEngines/matRad_ParticlePencilBeamEngineAbstract.m

```diff
@@ -0,0 +1,902 @@
+            bixel.energyIx = energyIx;
+            bixel.baseData = this.machine.data(energyIx);
+            
+            bixel.rangeShifter = currRay.rangeShifter(k);
+            
+            bixel.SSD = currRay.SSD;
+
+            bixel.radDepthOffset = currRay.radDepthOffset;
+
+            % Compute initial spotWidth
+            bixel.sigmaIniSq = currRay.sigmaIni(k).^2;
+
+            % Apply beam modifiers
+            bixel = this.getBeamModifiers(bixel);
+           
+            %Gets bixel.ix (voxel indices) and bixel.subIx (logical
+            %indices to be used)  after cutoff. Storing these allows us to
+            %use indexing for performance and avoid to many copies
+            bixel = this.getBixelIndicesOnRay(bixel,currRay);
+
+            if isempty(bixel.ix)
+                return;
+            end     
+
+            % Get quantities 1:1 from ray. Here we trust Matlab's memory
+            % management to not copy the arrays until they are modified.
+            % This allows us to efficiently access them by indexing in the
+            % bixel computation
+            bixel.radialDist_sq = currRay.radialDist_sq(bixel.subRayIx);
+            bixel.radDepths = currRay.radDepths(bixel.subRayIx);
+            if this.calcBioDose
+                bixel.vTissueIndex = currRay.vTissueIndex(bixel.subRayIx);
+            end
+        
+        end
+
+        function X = interpolateKernelsInDepth(this,bixel)
+            baseData = bixel.baseData;
+            
+            depths = baseData.depths;
+
+            % add potential offset
+            if isfield(baseData,'offset')
+                depths = depths + baseData.offset - bixel.radDepthOffset;
+            end
+            
+            % calculate particle dose for bixel k on ray j of beam i
+            % convert from MeV cm^2/g per primary to Gy mm^2 per 1e6 primaries
+            conversionFactor = 1.6021766208e-02;
+            
+            %Find all values we need to interpolate
+            X.Z = conversionFactor*baseData.Z;
+            
+            %Lateral Kernel Model
+            switch this.lateralModel
+                case 'single'
+                    X.sigma  = baseData.sigma;
+                case 'double'
+                    X.sigma1 = baseData.sigma1;
+                    X.sigma2 = baseData.sigma2;
+                    X.weight = baseData.weight;
+                case 'multi'
+                    X.weightMulti = baseData.weightMulti;
+                    X.sigmaMulti = baseData.sigmaMulti;
+                otherwise
+                    %Sanity check
+                    matRad_cfg = MatRad_Config.instance();
+                    matRad_cfg.dispError('Invalid Lateral Model');
+            end
+                   
+            % LET
+            if this.calcLET
+                X.LET = baseData.LET;
+            end
+
+            % bioDose
+            if this.calcBioDose
```

Comment:
I think this interpolation only makes sense if there is an alpha/beta kernel, but that is not the case for proton LET based models. If this interpolation is specific to the LEM model only we could just let the bio model class take care of it

Reply:
This is resolved with a temporary solution until @remocristoforetti updates the Biological model infrastructure